### PR TITLE
Refactored to use idiomatic package names.

### PR DESCRIPTION
**Pull request by empijei**
See more here: https://github.com/golang/go/wiki/CodeReviewComments#package-names

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ const (
 	refreshQueueSize  = 2048
 )
 
+// Server is a caching dns proxy that upgrades DNS to DNS over TLS.
```
Comment (mikispag) [inline server/server.go]: Nit: Make "dns" uppercase. Use "DNS-over-TLS" for coherence.

File: proxy/server_test.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func (f fakeListener) dialer() func(addr string, c *tls.Config) (net.Conn, error
 func setupTestServer(tb testing.TB, cacheSize int, responder func(q string) string) (exchanger func(logPrefix string), cleanup func()) {
 	tb.Helper()
 	const (
-		raddr    = "gopher.empijei:853"
+		raddr    = "gopher.empijei:53"
```
Comment (mikispag) [inline proxy/server_test.go]: Shouldn't it have :853 as it is a DNS-over-TLS upstream?

File: server/server.go (hunk 1 above)
Comment (empijei) [reply server/server.go]: Done.

File: proxy/server_test.go (hunk 2 above)
Comment (empijei) [reply proxy/server_test.go]: Done.

**Pull request merged**